lighthouse-core/lib/tracehouse/main-thread-tasks.js

```diff
@@ -262,10 +262,42 @@ class MainThreadTasks {
             // It's ending at traceEndTs, it means we were missing the end event. We'll truncate it to the parent.
             nextTask.endTime = currentTask.endTime;
             nextTask.duration = nextTask.endTime - nextTask.startTime;
+          } else if (
+            nextTask.startTime - currentTask.startTime < 1000 &&
+            !currentTask.children.length
+          ) {
+            // The parent started less than 1ms before the child, we'll let it slide by swapping the two,
```

Comment:
should this be the opposite (parent<->child)?

Reply:
maybe we should give up using 'parent' and 'child' in these swapping contexts 😆 this is written in terms of the actual structure of the trace events, not what we the current object structure might reflect. 

maybe we take the same variable name approach from before and just slap `actual` before these two?